[Bug] GUI startup failure caused by Infobar with wxPython 4.1

**Describe the bug**
Latest changes with `gui_core/infobar.py` cases failure to start up gui using wxPython 4.1. See traceback below.

```
Traceback (most recent call last):
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/wxgui.py", line 105, in OnInit
    mainframe = GMFrame(parent=None, id=wx.ID_ANY,
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/lmgr/frame.py", line 152, in __init__
    self.notebook = self._createNoteBook()
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/lmgr/frame.py", line 321, in _createNoteBook
    self.datacatalog = DataCatalog(
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/datacatalog/catalog.py", line 55, in __init__
    self.infoBar = InfoBar(self)
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/gui_core/infobar.py", line 51, in __init__
    self.subSizerText.Add(self._icon, wx.SizerFlags().Centre().Border())
wx._core.wxAssertionError: C++ assertion ""!m_containingSizer"" failed at /Users/robind/projects/bb2/dist-osx-py39/build/ext/wxWidgets/src/common/wincmn.cpp(2471) in SetContainingSizer(): Adding a window already in a sizer, detach it first!
OnInit returned false, exiting...
[...]
Launching <wxpython> GUI in the background, please wait...
Traceback (most recent call last):
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/wxgui.py", line 105, in OnInit
    mainframe = GMFrame(parent=None, id=wx.ID_ANY,
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/lmgr/frame.py", line 152, in __init__
    self.notebook = self._createNoteBook()
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/lmgr/frame.py", line 321, in _createNoteBook
    self.datacatalog = DataCatalog(
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/datacatalog/catalog.py", line 55, in __init__
    self.infoBar = InfoBar(self)
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/gui_core/infobar.py", line 52, in __init__
    self.subSizerText.Add(self._text, 1, wx.ALIGN_CENTER_VERTICAL)
wx._core.wxAssertionError: C++ assertion ""!m_containingSizer"" failed at /Users/robind/projects/bb2/dist-osx-py39/build/ext/wxWidgets/src/common/wincmn.cpp(2471) in SetContainingSizer(): Adding a window already in a sizer, detach it first!
OnInit returned false, exiting...
[...]
Launching <wxpython> GUI in the background, please wait...
Traceback (most recent call last):
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/wxgui.py", line 105, in OnInit
    mainframe = GMFrame(parent=None, id=wx.ID_ANY,
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/lmgr/frame.py", line 152, in __init__
    self.notebook = self._createNoteBook()
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/lmgr/frame.py", line 321, in _createNoteBook
    self.datacatalog = DataCatalog(
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/datacatalog/catalog.py", line 55, in __init__
    self.infoBar = InfoBar(self)
  File "/grass/dist.x86_64-apple-darwin19.6.0/gui/wxpython/gui_core/infobar.py", line 54, in __init__
    self.subSizerButtons.Add(self._button, wx.SizerFlags().Centre().Border())
wx._core.wxAssertionError: C++ assertion ""!m_containingSizer"" failed at /Users/robind/projects/bb2/dist-osx-py39/build/ext/wxWidgets/src/common/wincmn.cpp(2471) in SetContainingSizer(): Adding a window already in a sizer, detach it first!
```


**System description (please complete the following information):**

- GRASS GIS version 7.9.dev 

Python:
3.9.1 (v3.9.1:1e5d33e9b9, Dec  7 2020, 12:10:52) 
[Clang 6.0 (clang-600.0.57)]
4.1.1 osx-cocoa (phoenix) wxWidgets 3.1.5
<!---

- details about further software components
    - run `g.version -rge` in a GRASS GIS terminal session or check in
      the GUI menu "Help > About"
    - run `python3 -c "import sys, wx; print(sys.version); print(wx.version())"`
      to print the Python and wxPython version numbers
--->
version=7.9.dev
date=2020
revision=5f9f7920a
build_date=2020-12-21
build_platform=x86_64-apple-darwin19.6.0
build_off_t_size=8
libgis_revision=7d9a0da4f
libgis_date=2020-12-21T11:16:51+00:00
proj=7.2.0
gdal=3.2.0
geos=3.8.1
sqlite=3.33.0

**Additional context**
Adding something like:
```
        self.GetSizer().Detach(self._icon)
        self.GetSizer().Detach(self._text)
        self.GetSizer().Detach(self._button)
```

to `Infobar.__init__()` before the layout section could perhaps be a solution.

--- a/gui/wxpython/gui_core/infobar.py
+++ b/gui/wxpython/gui_core/infobar.py
@@ -44,17 +44,27 @@ class InfoBar(IB.InfoBar):
         self._text.SetForegroundColour(self._foreground_color)
         self._button.SetBackgroundColour(self._background_color)
 
-        # LAYOUT
+        # layout
+        self._unInitLayout()
         sizer = wx.BoxSizer(wx.VERTICAL)
-        self.subSizerText = wx.BoxSizer(wx.HORIZONTAL)
+        subSizerText = wx.BoxSizer(wx.HORIZONTAL)
         self.subSizerButtons = wx.BoxSizer(wx.HORIZONTAL)
-        self.subSizerText.Add(self._icon, wx.SizerFlags().Centre().Border())
-        self.subSizerText.Add(self._text, 1, wx.ALIGN_CENTER_VERTICAL)
+        subSizerText.Add(self._icon, wx.SizerFlags().Centre().Border())
+        subSizerText.Add(self._text, 1, wx.ALIGN_CENTER_VERTICAL)
         self.subSizerButtons.AddStretchSpacer()
         self.subSizerButtons.Add(self._button, wx.SizerFlags().Centre().Border())
-        sizer.Add(self.subSizerText, wx.SizerFlags().Expand())
+        sizer.Add(subSizerText, wx.SizerFlags().Expand())
         sizer.Add(self.subSizerButtons, wx.SizerFlags().Expand())
         self.SetSizer(sizer)
+
+    def _unInitLayout(self):
+        sizer = self.GetSizer()
+        children = sizer.GetChildren()
+        for i in reversed(range(len(children))):
+            if children[i].IsSpacer():
+                sizer.Remove(i)
+            else:
+                sizer.Detach(i)
 
     def ShowMessage(self, message, icon, buttons=None):
         """Show message with buttons (optional).

